src/globus_cli/parsing/commands.py

```diff
@@ -51,6 +52,13 @@ def __init__(self, *args, **kwargs):
             kwargs["help"] = helptext.format(
                 AUTOMATIC_ACTIVATION=self.AUTOMATIC_ACTIVATION_HELPTEXT
             )
+        if "context_settings" not in kwargs:
+            kwargs["context_settings"] = {}
+        if "max_content_width" not in kwargs["context_settings"]:
+            term_size = get_terminal_size(fallback=(80, 20))
+            kwargs["context_settings"]["max_content_width"] = int(
+                0.8 * term_size.columns
+            )
```

Comment:
One thing which surprised me with this change is that `globus transfer -h | cat` became _narrower_.
I'm wondering if we should "widen out" if the terminal width is small, which will also cover the fallback case.

So, something like (and don't mind exactly how I write this please 😉 ):
```python
cols = get_terminal_size(fallback=...).columns
if cols < 100: cols
else: cols * 0.8
```

Does that seem nice?

Reply:
Oh, right. The fallback will take over for the pipe case, so we don't want to again take 80% of that.

I also wonder whether about cases where `get_terminal_size` could fail; I figure it doesn't help to catch those.

Reply:
Just looking at the [py3.10 source for get_terminal_size](https://github.com/python/cpython/blob/251104f12d0cebb997f26f8ca3dac48202c7456a/Lib/shutil.py#L1378) I think this is already an "error handling variant" of `os.get_terminal_size(sys.stdout.fileno())`.

I would assume that it works as expected and we can handle any bug reports if someone finds a context where it does not. It doesn't look like there have been changes there since py3.5, so I bet it's pretty solid.